experiment/telegram/telegram_otherwise.go

```diff
@@ -0,0 +1,184 @@
+// +build !cgo
+
+// Package telegram contains the Telegram network experiment. This file
+// in particular is a pure-Go implementation of that.
+//
+// See https://github.com/ooni/spec/blob/master/nettests/ts-020-telegram.md.
+package telegram
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ooni/netx/handlers"
+	"github.com/ooni/netx/httpx"
+	"github.com/ooni/probe-engine/experiment"
+	"github.com/ooni/probe-engine/experiment/handler"
+	"github.com/ooni/probe-engine/log"
+	"github.com/ooni/probe-engine/model"
+	"github.com/ooni/probe-engine/session"
+)
+
+const (
+	testName    = "telegram"
+	testVersion = "0.0.2"
+)
+
+// Config contains the experiment config.
+type Config struct{}
+
+// TestKeys contains telegram test keys.
+type TestKeys struct {
+	TelegramHTTPBlocking bool   `json:"telegram_http_blocking"`
+	TelegramTCPBlocking  bool   `json:"telegram_tcp_blocking"`
+	TelegramWebFailure   string `json:"telegram_web_failure"`
+	TelegramWebStatus    string `json:"telegram_web_status"`
```

Comment:
Where is the `tcp_connect` list populated with the result items?

Reply:
It is not. As mentioned in the PR description, this is a minimal implementation of Telegram that does not include any data format like `tcp_connect`. Those should be computed using the data stream from `netx` using the same code for all tests. There is no point in adding them in this stage of the PR, except making the diff long and difficult to read.

Reply:
Correction: there was no point for me to adding them at the beginning of the review of the PR except making the diff more long and difficult to read for you. That said, I am open to add this code in here if you wish. My preference, however, would be to do that at a later stage when we have a bunch of tests, so I can better understand what is required.

Reply:
Hum, I think it would be useful to have it done as part of this PR so that we can better understand how everything fits together and validate that the result of this test matches up with the result of the measurement-kit based one (i.e. I can setup some rules on my machine to simulate some forms of blocking and check that both tests produce similar results).

Beyond just looking at the diff I think it's most useful to actually test the code and compare the two code-bases against each other.

Reply:
As discussed in video chat, this functionality will be added later and we will proceed piecemeal for now.

Reply:
ok, will implement the full test then